rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -0,0 +1,416 @@
+
+		to := from + defaultFetchBlockRange - 1
+		if to > latestConfirmed {
+			to = latestConfirmed
+		}
+
+		logs, err := s.client.fetchRollupEventsInRange(s.ctx, from, to)
+		if err != nil {
+			log.Error("failed to fetch rollup events in range", "from block", from, "to block", to, "err", err)
+			return
+		}
+
+		if err := s.parseAndUpdateRollupEventLogs(logs, to); err != nil {
+			log.Error("failed to parse and update rollup event logs", "err", err)
+			return
+		}
+
+		s.latestProcessedBlock = to
+	}
+}
+
+func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endBlockNumber uint64) error {
+	for _, vLog := range logs {
+		switch vLog.Topics[0] {
+		case s.l1CommitBatchEventSignature:
+			event := &L1CommitBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "CommitBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack commit rollup event log, err: %w", err)
+			}
+			batchIndex := event.BatchIndex.Uint64()
+			log.Trace("found new CommitBatch event", "batch index", batchIndex)
+
+			chunkBlockRanges, err := s.getChunkRanges(batchIndex, &vLog)
+			if err != nil {
+				return fmt.Errorf("failed to get chunk ranges, err: %w", err)
+			}
+			rawdb.WriteBatchChunkRanges(s.db, batchIndex, chunkBlockRanges)
+
+		case s.l1RevertBatchEventSignature:
+			event := &L1RevertBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "RevertBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack revert rollup event log, err: %w", err)
+			}
+			batchIndex := event.BatchIndex.Uint64()
+			log.Trace("found new RevertBatch event", "batch index", batchIndex)
+
+			rawdb.DeleteBatchChunkRanges(s.db, batchIndex)
+
+		case s.l1FinalizeBatchEventSignature:
+			event := &L1FinalizeBatchEvent{}
+			if err := UnpackLog(s.scrollChainABI, event, "FinalizeBatch", vLog); err != nil {
+				return fmt.Errorf("failed to unpack finalized rollup event log, err: %w", err)
+			}
+			batchIndex := event.BatchIndex.Uint64()
+			log.Trace("found new FinalizeBatch event", "batch index", batchIndex)
+
+			parentBatchMeta, chunks, err := s.getLocalInfoForBatch(batchIndex)
+			if err != nil {
+				return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", batchIndex, err)
+			}
+
+			startBlock, endBlock, finalizedBatchMeta, err := validateBatch(event, parentBatchMeta, chunks)
+			if err != nil {
+				return fmt.Errorf("fatal: validateBatch failed: finalize event: %v, err: %w", event, err)
+			}
+
+			batchWriter := s.db.NewBatch()
+			rawdb.WriteFinalizedL2BlockNumber(batchWriter, endBlock)
+			rawdb.WriteBlockNumberToBatchIndex(batchWriter, startBlock, endBlock, batchIndex)
+			rawdb.WriteFinalizedBatchMeta(batchWriter, batchIndex, finalizedBatchMeta)
+			if err := batchWriter.Write(); err != nil {
+				log.Crit("Failed to write batch to database", "err", err)
+			}
+
+			if batchIndex%100 == 0 {
+				log.Info("finalized batch progress", "batch index", batchIndex, "finalized l2 block height", endBlock)
```

Comment:
How about we also print an update once every 5 minutes or so? I feel it would be nice to see more logs when the node is synced, waiting for 100 batches takes a long time.

Reply:
changed to print by a `Ticker` in `Start()` in [88eed96](https://github.com/scroll-tech/go-ethereum/pull/515/commits/88eed967ad470daf15d7f952355e296d31895a01)